hazelcast/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -378,6 +379,9 @@ public void start() {
             clusterService.start(configuredListeners);
             clientClusterViewListenerService.start();
 
+            configuredListeners.stream().filter(ClientConnectionProcessListener.class::isInstance)
```

Comment:
I see that the listener is defined in the impl package, but the configured listeners are meant to be used as an aggregate for the listeners we gave as a public API. I don't think `ClientConnectionProcessListener` fits there. If it is internal, it should be added via internal methods.

Reply:
As discussed in DM, we keep it as-is for now, just adding a comment that this is a private API